tests/test_api.py

```diff
@@ -314,3 +326,165 @@ def f():
         while x > 0:
             x = x - 1
         return x
+
+
+#####################################################
+# Test convert_arg_init functions used by to_device #
+#####################################################
+
+device_type = 'cpu'
+# device_type = 'cuda'  # Uncomment to run on the gpu
+
+
+@pytest.fixture(params=[
+    pytest.param(('nnvm', {'target': 'cpu', 'device_id': 0}), id='nnvm-cpu'),
+    pytest.param(('relay', {'target': 'cpu', 'device_id': 0}), id='relay-cpu'),
+    pytest.param(('pytorch', {'device': 'cpu'}), id='pytorch-cpu')])
+def backend_opt(request):
+    name, options = request.param
+    return (name, options)
+
+
+__W = np.array(
+    [
+        [0.22490819, -0.71516967, -0.6626606],
+        [-0.14859799, 0.924496, 0.4032511]
+    ]
+)
```

Comment:
I don't know if the exact values for this matrix are important, but there are a few premade matrices in `tests.common`. You can use e.g. `tests.common.MA(2, 3)` instead of `__W`, for a 2x3 matrix.